frontend/src/components/MinioArtifactLink.tsx

```diff
@@ -1,30 +1,36 @@
 import * as React from 'react';
-import { StoragePath, StorageService } from '../lib/WorkflowParser';
+import { StorageService } from '../lib/WorkflowParser';
 import { S3Artifact } from '../../third_party/argo-ui/argo_template';
-
-const artifactApiUri = ({ source, bucket, key }: StoragePath) =>
-  'artifacts/get' + `?source=${source}&bucket=${bucket}&key=${encodeURIComponent(key)}`;
+import { generateArtifactUrl } from '../lib/Utils';
+import { Apis } from '../lib/Apis';
 
 /**
- * A component that renders an artifact link.
+ * A component that renders a preview to an artifact with a link to the full content.
  */
 const MinioArtifactLink: React.FC<S3Artifact> = s3artifact => {
-  if (!s3artifact || !s3artifact.key || !s3artifact.bucket) {
+  const { key, bucket, endpoint } = s3artifact || {};
+  if (!key || !bucket) {
     return null;
   }
 
-  const { key, bucket, endpoint } = s3artifact;
+  const peek = 100;
+  const encodedKey = encodeURIComponent(key);
   const source = endpoint === 's3.amazonaws.com' ? StorageService.S3 : StorageService.MINIO;
-  const linkText = `${source.toString()}://${bucket}/${key}`;
+  const linkText = `${source.toString()}://${bucket}/${encodedKey}`;
+  const url = generateArtifactUrl(source, bucket, encodedKey);
+  const [content, setContent] = React.useState(linkText);
+
+  React.useEffect(() => {
+    Apis.readFile({ source, bucket, key }, peek).then(
+      data => setContent(data.length > peek ? `${data} ...` : data),
+      error => setContent(error),
```

Comment:
Directly showing error message doesn't look good, also there's no way to retry.
Two options I can think of:
* Don't show the preview if it failed
* Show a written error message and log the root cause in console